{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "legendary-scotland",
   "metadata": {},
   "source": [
    "## Use BLAST2 to check for sequences\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "controversial-conviction",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import time\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pathlib\n",
    "from io import StringIO\n",
    "from Bio import SeqIO\n",
    "from Bio.Cluster import distancematrix\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio.Blast import NCBIWWW\n",
    "from Bio.Blast import NCBIXML\n",
    "import subprocess\n",
    "from recan.simgen import Simgen\n",
    "from matplotlib import pyplot as plt\n",
    "import seaborn as sns\n",
    "from os import listdir\n",
    "from os.path import isfile, join"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "difficult-saver",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.options.display.max_seq_items = 2000"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "individual-values",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "```\n",
    "makeblastdb -in SRR5722761_final.contigs.fa -dbtype nucl -parse_seqids -out /mnt/1TB_ssd/Data/BLAST/asm_db/SRR5722761_megahit_default\n",
    "```\n",
    "\n",
    "\n",
    "```\n",
    "export BLASTDB=/mnt/1TB_ssd/Data/BLAST/asm_db\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "hydraulic-blood",
   "metadata": {},
   "outputs": [],
   "source": [
    "PROJECT_CODE='PRJNA396502'\n",
    "\n",
    "BASE_PATH=f'/mnt/1TB_0/Data/Assembly/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "painful-cleveland",
   "metadata": {},
   "outputs": [],
   "source": [
    "SRA_LIST=['SRR5885860']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "parliamentary-spine",
   "metadata": {},
   "source": [
    "### BSL3/4 organisms and vector search and Bat CoV search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cultural-cause",
   "metadata": {},
   "outputs": [],
   "source": [
    "ORGANISM_CSVS=['/mnt/1TB_0/Data/fasta/organisms_all.csv', '/mnt/1TB_0/Data/fasta/cov_common_name.csv']\n",
    "CSV_PATHS=['/mnt/1TB_0/Data/fasta/complete_nucleotide/','/mnt/1TB_0/Data/fasta/common_name_fasta/']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ranging-block",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_db(asm_path, sra_code):\n",
    "    cmd = f\"makeblastdb -in {asm_path+sra_code}_final.contigs.fa -dbtype nucl -parse_seqids -out /mnt/1TB_ssd/Data/BLAST/asm_db/{sra_code}_megahit_default\"\n",
    "    subprocess.check_call(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "gothic-discussion",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cmd = \"export BLASTDB=/mnt/1TB_ssd/Data/BLAST/asm_db\"\n",
    "subprocess.check_call(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "bound-internship",
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_path(sequences, fasta_path):\n",
    "    for fname in sequences:\n",
    "        if not os.path.isfile(fasta_path+fname):\n",
    "            print(f'File non existant: {fasta_path}{fname}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "virtual-freeze",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_data(df):\n",
    "    sequences=df['File_Name']\n",
    "    accessions=df['Accession']\n",
    "    assert len(sequences)==len(accessions)\n",
    "    return sequences, accessions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "optical-tribune",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_blast2(sequences, accessions, sra_code, fasta_path, out_file_path):\n",
    "    for g, asc in zip(sequences, accessions):\n",
    "        out_file=f'{sra_code}_{asc}_BLAST2.txt'\n",
    "        db=f'{sra_code}_megahit_default'\n",
    "        cmd = f\"blastn -query {fasta_path+g} -db {db} -num_threads 6 -out {out_file_path+out_file} -evalue 0.001 -max_target_seqs 100 -perc_identity 90 -max_hsps 10\"\n",
    "        subprocess.check_call(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "transsexual-monaco",
   "metadata": {},
   "outputs": [],
   "source": [
    "def cleanup(sequences, accessions, sra_code, out_file_path):\n",
    "    del_list=[]\n",
    "    for g, asc in zip(sequences, accessions):\n",
    "        out_file=f'{sra_code}_{asc}_BLAST2.txt'\n",
    "        with open(out_file_path+out_file) as myfile:\n",
    "            if '***** No hits found *****' in myfile.read():\n",
    "                del_list.append(out_file)\n",
    "    for f in del_list:\n",
    "        if os.path.exists(out_file_path+f):\n",
    "            os.remove(out_file_path+f)\n",
    "        else:\n",
    "            print(f\"{f} does not exist\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "heard-briefing",
   "metadata": {},
   "outputs": [],
   "source": [
    "def blast_workflow(sequences, accessions,csv_path, search_name, rebuild_db):\n",
    "    match_dict={}\n",
    "    for sra_code in SRA_LIST:\n",
    "        start_time=time.time()\n",
    "        print(f'Running blast2 on {sra_code}')\n",
    "        matches=[]\n",
    "        data_path=f'{BASE_PATH}{PROJECT_CODE}/{sra_code}/'\n",
    "        asm_path=data_path+'megahit_default/'\n",
    "        b_path=data_path+f'megahit_asm_BLAST2/'\n",
    "        pathlib.Path(b_path).mkdir(exist_ok=True)\n",
    "        out_file_path=data_path+f'megahit_asm_BLAST2/{search_name}/'\n",
    "        pathlib.Path(out_file_path).mkdir(exist_ok=True)\n",
    "        if rebuild_db:\n",
    "            make_db(asm_path, sra_code)\n",
    "        elif not os.path.isfile(f'/mnt/1TB_ssd/Data/BLAST/asm_db/{sra_code}_megahit_default.ndb'):\n",
    "            make_db(asm_path, sra_code)\n",
    "        run_blast2(sequences, accessions, sra_code, csv_path, out_file_path)\n",
    "        cleanup(sequences, accessions, sra_code, out_file_path)\n",
    "        matchfiles = [f for f in listdir(out_file_path) if isfile(join(out_file_path, f))]\n",
    "        for f in matchfiles:\n",
    "            asc=f.split('_',1)[1]\n",
    "            asc=asc.split('_BLAST2.txt')[0]\n",
    "            try:\n",
    "                idx = accessions.tolist().index(asc.strip())\n",
    "                match=sequences[idx]\n",
    "                matches.append(match)\n",
    "            except ValueError as e:\n",
    "                print(f'accessions:{str(accessions.tolist())}')\n",
    "                print(f'asc: {asc}, file: {f} error: {e}')\n",
    "                raise ValueError\n",
    "        match_dict[sra_code]=matches\n",
    "        print(f'Elapsed: {time.time()-start_time}')\n",
    "    return match_dict\n",
    "                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "israeli-gathering",
   "metadata": {},
   "outputs": [],
   "source": [
    "def main_worflow(organism_csv, csv_path, search_name, rebuild_db):\n",
    "    result_list=[]\n",
    "    df=pd.read_csv(organism_csv)\n",
    "    sequences, accessions=get_data(df)\n",
    "    match_dict=blast_workflow(sequences, accessions,csv_path, search_name, rebuild_db)\n",
    "    result_list.append(match_dict)\n",
    "    return result_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "daily-district",
   "metadata": {},
   "source": [
    "## BSL organism hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "bottom-express",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running blast2 on SRR5885860\n",
      "Elapsed: 326.1996157169342\n"
     ]
    }
   ],
   "source": [
    "bsl_result_list=main_worflow(ORGANISM_CSVS[0],CSV_PATHS[0], search_name='bsl', rebuild_db=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "cosmetic-stream",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'SRR5885860': ['NC_000007_14_CFTR_gene.fa',\n",
       "   'JN950623_1_Mus_musculus_targeted_non-conditional_lacZ-tagged_mutant_allele_Herc1_tm1e_EUCOMM_Wtsi_transgenic.fa',\n",
       "   'KJ473820_1_BtPa-BetaCoV_GD2013_complete_genome.fa',\n",
       "   'MK787297_1_cloning_vector_pR6K-KCA.fa',\n",
       "   'MK280359_1_Homo_sapiens_lncAB370_3_lncRNA_gene_complete_sequence.fa',\n",
       "   'MK279923_1_Homo_sapiens_lncAB599_3_lncRNA_gene_complete_sequence.fa',\n",
       "   'MK280367_1_Homo_sapiens_lncAB371_6_lncRNA_gene_complete_sequence.fa',\n",
       "   'MN996867_1_Cloning_vector_pcDNA3_1_complete_sequence.fa',\n",
       "   'NC_006549_1_Singapore_grouper_iridovirus_complete_genome.fa',\n",
       "   'MN611520_1_Pipistrellus_abramus_bat_coronavirus_HKU5-related_isolate_BY140568_complete_genome.fa',\n",
       "   'NC_000004_12_ABCG2_gene.fa',\n",
       "   'NC_000005_10_TERT_gene.fa',\n",
       "   'EU410304_1_Synthetic_Vaccinia_virus_clone_GLV-1h68_complete_genome.fa',\n",
       "   'NC_009020_1_HKU5_complete_genome.fa',\n",
       "   'MF164268_1_Homo_sapiens_clone_BAC_JH12_genomic_sequence.fa']}]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bsl_result_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "latter-prior",
   "metadata": {},
   "source": [
    "### BSL list of organisms we are shearching for"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "resistant-cocktail",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Accession</th>\n",
       "      <th>File_Name</th>\n",
       "      <th>category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bacteriophage_S13_complete_genome</td>\n",
       "      <td>AF274751_1</td>\n",
       "      <td>AF274751_1_Bacteriophage_S13_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Human_endogenous_retrovirus_H_HERV-H_env62_pro...</td>\n",
       "      <td>AJ289709_1</td>\n",
       "      <td>AJ289709_1_Human_endogenous_retrovirus_H_HERV-...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Porcine_hemagglutinating_encephalomyelitis_vir...</td>\n",
       "      <td>DQ011855_1</td>\n",
       "      <td>DQ011855_1_Porcine_hemagglutinating_encephalom...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Bat_SARS_coronavirus HKU3-1_complete_genome</td>\n",
       "      <td>DQ022305_2</td>\n",
       "      <td>DQ022305_2_Bat_SARS_coronavirus HKU3-1_complet...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Coliphage_ID45_complete_genome</td>\n",
       "      <td>DQ079883_1</td>\n",
       "      <td>DQ079883_1_Coliphage_ID45_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Coliphage_WA11_complete_genome</td>\n",
       "      <td>DQ079895_1</td>\n",
       "      <td>DQ079895_1_Coliphage_WA11_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>bat_SARS_coronavirus_HKU3-2_complete_genome</td>\n",
       "      <td>DQ084199_1</td>\n",
       "      <td>DQ084199_1_bat_SARS_coronavirus_HKU3-2_complet...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>bat_SARS_coronavirus_HKU3-3_complete_genome</td>\n",
       "      <td>DQ084200_1</td>\n",
       "      <td>DQ084200_1_bat_SARS_coronavirus_HKU3-3_complet...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Synthetic_Vaccinia_virus_clone_GLV-1h68_comple...</td>\n",
       "      <td>EU410304_1</td>\n",
       "      <td>EU410304_1_Synthetic_Vaccinia_virus_clone_GLV-...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Cote_d-Ivoire_ebolavirus_complete_genome</td>\n",
       "      <td>FJ217162_1</td>\n",
       "      <td>FJ217162_1_Cote_d-Ivoire_ebolavirus_complete_g...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>XMRV-like_mouse_endogenous_retrovirus_mERV-XL_...</td>\n",
       "      <td>JF714652_1</td>\n",
       "      <td>JF714652_1_XMRV-like_mouse_endogenous_retrovir...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Bat_SARS-like_coronavirus_Rs3367_complete_genome</td>\n",
       "      <td>KC881006_1</td>\n",
       "      <td>KC881006_1_Bat_SARS-like_coronavirus_Rs3367_co...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Bat_SARS-like_coronavirus_WIV1_complete_genome</td>\n",
       "      <td>KF367457_1</td>\n",
       "      <td>KF367457_1_Bat_SARS-like_coronavirus_WIV1_comp...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Human_coronavirus_HKU1_strain_HKU1-human-USA-H...</td>\n",
       "      <td>KF686346_1</td>\n",
       "      <td>KF686346_1_Human_coronavirus_HKU1_strain_HKU1-...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Enterobacteria_phage_MED1_complete_genome</td>\n",
       "      <td>KJ997912_1</td>\n",
       "      <td>KJ997912_1_Enterobacteria_phage_MED1_complete_...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Human_adenovirus_12_isolate_GyK010_complete_ge...</td>\n",
       "      <td>KX868289_2</td>\n",
       "      <td>KX868289_2_Human_adenovirus_12_isolate_GyK010_...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Mengo_virus_isolate_M_complete_genome</td>\n",
       "      <td>L22089_1</td>\n",
       "      <td>L22089_1_Mengo_virus_isolate_M_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Woodchuck_hepatitis_virus_WHV_complete_genome_...</td>\n",
       "      <td>M19183_1</td>\n",
       "      <td>M19183_1_Woodchuck_hepatitis_virus_WHV_complet...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Bat_SARS-like_coronavirus_isolate_bat-SL-CoVZC...</td>\n",
       "      <td>MG772933_1</td>\n",
       "      <td>MG772933_1_Bat_SARS-like_coronavirus_isolate_b...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Porcine_deltacoronavirus_isolate_CH_JXJGS01_20...</td>\n",
       "      <td>MK625641_1</td>\n",
       "      <td>MK625641_1_Porcine_deltacoronavirus_isolate_CH...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Swine_acute_diarrhea_syndrome_coronavirus_stra...</td>\n",
       "      <td>MK651076_1</td>\n",
       "      <td>MK651076_1_Swine_acute_diarrhea_syndrome_coron...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Rabies_lyssavirus_isolate_19RS359_complete_genome</td>\n",
       "      <td>MK920923_1</td>\n",
       "      <td>MK920923_1_Rabies_lyssavirus_isolate_19RS359_c...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>RaTG13_complete_genome</td>\n",
       "      <td>MN996532_1</td>\n",
       "      <td>MN996532_1_RaTG13_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Cloning_vector_pcDNA3_1_complete_sequence</td>\n",
       "      <td>MN996867_1</td>\n",
       "      <td>MN996867_1_Cloning_vector_pcDNA3_1_complete_se...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Friend_murine_leukemia_virus_FB29_complete_genome</td>\n",
       "      <td>NC_001362_1</td>\n",
       "      <td>NC_001362_1_Friend_murine_leukemia_virus_FB29_...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Japanese_encephalitis_virus_genome</td>\n",
       "      <td>NC_001437_1</td>\n",
       "      <td>NC_001437_1_Japanese_encephalitis_virus_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Avian_infectious_bronchitis_virus_complete_genome</td>\n",
       "      <td>NC_001451_1</td>\n",
       "      <td>NC_001451_1_Avian_infectious_bronchitis_virus_...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Human_adenovirus_A_complete_genome</td>\n",
       "      <td>NC_001460_1</td>\n",
       "      <td>NC_001460_1_Human_adenovirus_A_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Dengue_virus_2_complete_genome</td>\n",
       "      <td>NC_001474_2</td>\n",
       "      <td>NC_001474_2_Dengue_virus_2_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Dengue_virus_1_complete_genome</td>\n",
       "      <td>NC_001477_1</td>\n",
       "      <td>NC_001477_1_Dengue_virus_1_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Moloney_murine_sarcoma_virus_complete_genome</td>\n",
       "      <td>NC_001502_1</td>\n",
       "      <td>NC_001502_1_Moloney_murine_sarcoma_virus_compl...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Simian_immunodeficiency_virus_complete_genome</td>\n",
       "      <td>NC_001549_1</td>\n",
       "      <td>NC_001549_1_Simian_immunodeficiency_virus_comp...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Marburg_marburgvirus_isolate_Marburg_virus-H.s...</td>\n",
       "      <td>NC_001608_3</td>\n",
       "      <td>NC_001608_3_Marburg_marburgvirus_isolate_Marbu...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>African_swine_fever_virus_strain_BA71V_complet...</td>\n",
       "      <td>NC_001659_2</td>\n",
       "      <td>NC_001659_2_African_swine_fever_virus_strain_B...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Simian_virus_40_complete_genome</td>\n",
       "      <td>NC_001669_1</td>\n",
       "      <td>NC_001669_1_Simian_virus_40_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Hendra_virus_complete_genome</td>\n",
       "      <td>NC_001906_3</td>\n",
       "      <td>NC_001906_3_Hendra_virus_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>canine_distemper_virus_complete_genome</td>\n",
       "      <td>NC_001921_1</td>\n",
       "      <td>NC_001921_1_canine_distemper_virus_complete_ge...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Feline_infectious_peritonitis_virus_complete_g...</td>\n",
       "      <td>NC_002306_3</td>\n",
       "      <td>NC_002306_3_Feline_infectious_peritonitis_viru...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Human_coronavirus_229E_complete_genome</td>\n",
       "      <td>NC_002645_1</td>\n",
       "      <td>NC_002645_1_Human_coronavirus_229E_complete_ge...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Nipah_virus_complete_genome</td>\n",
       "      <td>NC_002728_1</td>\n",
       "      <td>NC_002728_1_Nipah_virus_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Porcine_epidemic_diarrhea_virus_complete_genome</td>\n",
       "      <td>NC_003436_1</td>\n",
       "      <td>NC_003436_1_Porcine_epidemic_diarrhea_virus_co...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Cowpox_virus_complete_genome</td>\n",
       "      <td>NC_003663_2</td>\n",
       "      <td>NC_003663_2_Cowpox_virus_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Goatpox_virus_Pellor_complete_genome</td>\n",
       "      <td>NC_004003_1</td>\n",
       "      <td>NC_004003_1_Goatpox_virus_Pellor_complete_geno...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Hepatitis_C_virus_genotype_1_complete_genome</td>\n",
       "      <td>NC_004102_1</td>\n",
       "      <td>NC_004102_1_Hepatitis_C_virus_genotype_1_compl...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Chikungunya_virus_complete_genome</td>\n",
       "      <td>NC_004162_2</td>\n",
       "      <td>NC_004162_2_Chikungunya_virus_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>SARS_coronavirus_Tor2_complete_genome</td>\n",
       "      <td>NC_004718_3</td>\n",
       "      <td>NC_004718_3_SARS_coronavirus_Tor2_complete_gen...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Crimean-Congo_hemorrhagic_fever_virus_segment_...</td>\n",
       "      <td>NC_005300_2</td>\n",
       "      <td>NC_005300_2_Crimean-Congo_hemorrhagic_fever_vi...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Crimean-Congo_hemorrhagic_fever_virus_segment_...</td>\n",
       "      <td>NC_005301_3</td>\n",
       "      <td>NC_005301_3_Crimean-Congo_hemorrhagic_fever_vi...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Crimean-Congo_hemorrhagic_fever_virus_segment_...</td>\n",
       "      <td>NC_005302_1</td>\n",
       "      <td>NC_005302_1_Crimean-Congo_hemorrhagic_fever_vi...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Human_coronavirus_OC43_strain_ATCC_VR-759_comp...</td>\n",
       "      <td>NC_006213_1</td>\n",
       "      <td>NC_006213_1_Human_coronavirus_OC43_strain_ATCC...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Human_herpesvirus_5_strain_Merlin_complete_genome</td>\n",
       "      <td>NC_006273_2</td>\n",
       "      <td>NC_006273_2_Human_herpesvirus_5_strain_Merlin_...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Singapore_grouper_iridovirus_complete_genome</td>\n",
       "      <td>NC_006549_1</td>\n",
       "      <td>NC_006549_1_Singapore_grouper_iridovirus_compl...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>HKU4_complete_genome</td>\n",
       "      <td>NC_009019_1</td>\n",
       "      <td>NC_009019_1_HKU4_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>HKU5_complete_genome</td>\n",
       "      <td>NC_009020_1</td>\n",
       "      <td>NC_009020_1_HKU5_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Zika_virus_complete_genome</td>\n",
       "      <td>NC_012532_1</td>\n",
       "      <td>NC_012532_1_Zika_virus_complete_genome.fa</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Middle_East_respiratory_syndrome-related_coron...</td>\n",
       "      <td>NC_019843_3</td>\n",
       "      <td>NC_019843_3_Middle_East_respiratory_syndrome-r...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>Hubei_picorna-like_virus_45_strain_spider13352...</td>\n",
       "      <td>NC_032993_1</td>\n",
       "      <td>NC_032993_1_Hubei_picorna-like_virus_45_strain...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>Vesicular_stomatitis_Indiana_virus_strain_98CO...</td>\n",
       "      <td>NC_038236_1</td>\n",
       "      <td>NC_038236_1_Vesicular_stomatitis_Indiana_virus...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>UNVERIFIED_African_swine_fever_virus_isolate_O...</td>\n",
       "      <td>NC_044948_1</td>\n",
       "      <td>NC_044948_1_UNVERIFIED_African_swine_fever_vir...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>SARS_CoV_2_Wuahn-Hu-1_complete_genome</td>\n",
       "      <td>NC_045512_2</td>\n",
       "      <td>NC_045512_2_SARS_CoV_2_Wuahn-Hu-1_complete_gen...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Murine_hepatitis_virus_strain_A59_complete_genome</td>\n",
       "      <td>NC_048217_1</td>\n",
       "      <td>NC_048217_1_Murine_hepatitis_virus_strain_A59_...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>Avian_avulavirus_1_isolate_chicken-Indonesia-I...</td>\n",
       "      <td>MN727300_2</td>\n",
       "      <td>MN727300_2_Avian_avulavirus_1_isolate_chicken-...</td>\n",
       "      <td>BSL_organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>Bat_coronavirus_BtCoV-133-2005_complete_genome</td>\n",
       "      <td>DQ648794_1</td>\n",
       "      <td>DQ648794_1_Bat_coronavirus_BtCoV-133-2005_comp...</td>\n",
       "      <td>bat_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>Tylonycteris_pachypus_bat_coronavirus_HKU4-rel...</td>\n",
       "      <td>MN611519_1</td>\n",
       "      <td>MN611519_1_Tylonycteris_pachypus_bat_coronavir...</td>\n",
       "      <td>bat_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>BtPa-BetaCoV_GD2013_complete_genome</td>\n",
       "      <td>KJ473820_1</td>\n",
       "      <td>KJ473820_1_BtPa-BetaCoV_GD2013_complete_genome.fa</td>\n",
       "      <td>bat_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>Pipistrellus_abramus_bat_coronavirus_HKU5-rela...</td>\n",
       "      <td>MN611520_1</td>\n",
       "      <td>MN611520_1_Pipistrellus_abramus_bat_coronaviru...</td>\n",
       "      <td>bat_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Homo_sapiens_lncAB371_6_lncRNA_gene_complete_s...</td>\n",
       "      <td>MK280367_1</td>\n",
       "      <td>MK280367_1_Homo_sapiens_lncAB371_6_lncRNA_gene...</td>\n",
       "      <td>mamalian_genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>Homo_sapiens_lncAB370_3_lncRNA_gene_complete_s...</td>\n",
       "      <td>MK280359_1</td>\n",
       "      <td>MK280359_1_Homo_sapiens_lncAB370_3_lncRNA_gene...</td>\n",
       "      <td>mamalian_genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>Homo_sapiens_clone_BAC_JH12_genomic_sequence</td>\n",
       "      <td>MF164268_1</td>\n",
       "      <td>MF164268_1_Homo_sapiens_clone_BAC_JH12_genomic...</td>\n",
       "      <td>mamalian_genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>Homo_sapiens_lncAB599_3_lncRNA_gene_complete_s...</td>\n",
       "      <td>MK279923_1</td>\n",
       "      <td>MK279923_1_Homo_sapiens_lncAB599_3_lncRNA_gene...</td>\n",
       "      <td>mamalian_genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Rattus_norvegicus_4_BAC_CH230-209E1_Childrens_...</td>\n",
       "      <td>AC105600_5</td>\n",
       "      <td>AC105600_5_Rattus_norvegicus_4_BAC_CH230-209E1...</td>\n",
       "      <td>mamalian_genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>Pig_DNA_sequence_from_clone_CH242-68L20_on_chr...</td>\n",
       "      <td>FP326762_5</td>\n",
       "      <td>FP326762_5_Pig_DNA_sequence_from_clone_CH242-6...</td>\n",
       "      <td>mamalian_genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>Mus_musculus_targeted_non-conditional_lacZ-tag...</td>\n",
       "      <td>JN950623_1</td>\n",
       "      <td>JN950623_1_Mus_musculus_targeted_non-condition...</td>\n",
       "      <td>mamalian_genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>Mus_musculus_BAC_clone_RP23-168H19_from_10_com...</td>\n",
       "      <td>AC123074_2</td>\n",
       "      <td>AC123074_2_Mus_musculus_BAC_clone_RP23-168H19_...</td>\n",
       "      <td>mamalian_genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>J01749_1_cloning_vector_pBR322</td>\n",
       "      <td>J01749_1</td>\n",
       "      <td>J01749_1_cloning_vector_pBR322.fa</td>\n",
       "      <td>cloning_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>KY926444_1_cloning_vector_pET-hsp12</td>\n",
       "      <td>KY926444_1</td>\n",
       "      <td>KY926444_1_cloning_vector_pET-hsp12.fa</td>\n",
       "      <td>cloning_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>MK787297_1_cloning_vector_pR6K-KCA</td>\n",
       "      <td>MK787297_1</td>\n",
       "      <td>MK787297_1_cloning_vector_pR6K-KCA.fa</td>\n",
       "      <td>cloning_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>MK908408_1_cloning_vector_pUC_3GLA_U6_1_Bbsl</td>\n",
       "      <td>MK908408_1</td>\n",
       "      <td>MK908408_1_cloning_vector_pUC_3GLA_U6_1_Bbsl.fa</td>\n",
       "      <td>cloning_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>MN996867_1_cloning_vector_pcDNA3_1_complete</td>\n",
       "      <td>MN996867_1</td>\n",
       "      <td>MN996867_1_cloning_vector_pcDNA3_1_complete.fa</td>\n",
       "      <td>cloning_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>U13856_1_pGEX-5X1_cloning_vector</td>\n",
       "      <td>U13856_1</td>\n",
       "      <td>U13856_1_pGEX-5X1_cloning_vector.fa</td>\n",
       "      <td>cloning_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>X65300_1_cloning_vector_pGEM_1</td>\n",
       "      <td>X65300_1</td>\n",
       "      <td>X65300_1_cloning_vector_pGEM_1.fa</td>\n",
       "      <td>cloning_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>NC_000001_11_RPS6KA1_gene</td>\n",
       "      <td>NC_000001_11_RPS6KA1_gene</td>\n",
       "      <td>NC_000001_11_RPS6KA1_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>NC_000004_12_ABCG2_gene</td>\n",
       "      <td>NC_000004_12_ABCG2_gene</td>\n",
       "      <td>NC_000004_12_ABCG2_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>NC_000004_12_FGF2_gene</td>\n",
       "      <td>NC_000004_12_FGF2_gene</td>\n",
       "      <td>NC_000004_12_FGF2_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>NC_000005_10_TERT_gene</td>\n",
       "      <td>NC_000005_10_TERT_gene</td>\n",
       "      <td>NC_000005_10_TERT_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>NC_000006_12_VEGFA_gene</td>\n",
       "      <td>NC_000006_12_VEGFA_gene</td>\n",
       "      <td>NC_000006_12_VEGFA_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>NC_000007_14_CAV1_gene</td>\n",
       "      <td>NC_000007_14_CAV1_gene</td>\n",
       "      <td>NC_000007_14_CAV1_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>NC_000007_14_CFTR_gene</td>\n",
       "      <td>NC_000007_14_CFTR_gene</td>\n",
       "      <td>NC_000007_14_CFTR_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>NC_000007_14_KCNH2_gene</td>\n",
       "      <td>NC_000007_14_KCNH2_gene</td>\n",
       "      <td>NC_000007_14_KCNH2_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>NC_000017_11_BIRC5_gene</td>\n",
       "      <td>NC_000017_11_BIRC5_gene</td>\n",
       "      <td>NC_000017_11_BIRC5_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>NC_000017_11_TP53_gene</td>\n",
       "      <td>NC_000017_11_TP53_gene</td>\n",
       "      <td>NC_000017_11_TP53_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>NC_000019_10_BAX_gene</td>\n",
       "      <td>NC_000019_10_BAX_gene</td>\n",
       "      <td>NC_000019_10_BAX_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>NC_000021_9_SOD1_gene</td>\n",
       "      <td>NC_000021_9_SOD1_gene</td>\n",
       "      <td>NC_000021_9_SOD1_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>NC_000068_8_Cebpb_gene</td>\n",
       "      <td>NC_000068_8_Cebpb_gene</td>\n",
       "      <td>NC_000068_8_Cebpb_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>NC_000069_7_Adar_gene</td>\n",
       "      <td>NC_000069_7_Adar_gene</td>\n",
       "      <td>NC_000069_7_Adar_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>NC_000074_7_Mt1_gene</td>\n",
       "      <td>NC_000074_7_Mt1_gene</td>\n",
       "      <td>NC_000074_7_Mt1_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>NC_000076_7_Ascl1_gene</td>\n",
       "      <td>NC_000076_7_Ascl1_gene</td>\n",
       "      <td>NC_000076_7_Ascl1_gene.fa</td>\n",
       "      <td>mamalian_expression_vector</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Name  \\\n",
       "0                   Bacteriophage_S13_complete_genome   \n",
       "1   Human_endogenous_retrovirus_H_HERV-H_env62_pro...   \n",
       "2   Porcine_hemagglutinating_encephalomyelitis_vir...   \n",
       "3         Bat_SARS_coronavirus HKU3-1_complete_genome   \n",
       "4                      Coliphage_ID45_complete_genome   \n",
       "5                      Coliphage_WA11_complete_genome   \n",
       "6         bat_SARS_coronavirus_HKU3-2_complete_genome   \n",
       "7         bat_SARS_coronavirus_HKU3-3_complete_genome   \n",
       "8   Synthetic_Vaccinia_virus_clone_GLV-1h68_comple...   \n",
       "9            Cote_d-Ivoire_ebolavirus_complete_genome   \n",
       "10  XMRV-like_mouse_endogenous_retrovirus_mERV-XL_...   \n",
       "11   Bat_SARS-like_coronavirus_Rs3367_complete_genome   \n",
       "12     Bat_SARS-like_coronavirus_WIV1_complete_genome   \n",
       "13  Human_coronavirus_HKU1_strain_HKU1-human-USA-H...   \n",
       "14          Enterobacteria_phage_MED1_complete_genome   \n",
       "15  Human_adenovirus_12_isolate_GyK010_complete_ge...   \n",
       "16              Mengo_virus_isolate_M_complete_genome   \n",
       "17  Woodchuck_hepatitis_virus_WHV_complete_genome_...   \n",
       "18  Bat_SARS-like_coronavirus_isolate_bat-SL-CoVZC...   \n",
       "19  Porcine_deltacoronavirus_isolate_CH_JXJGS01_20...   \n",
       "20  Swine_acute_diarrhea_syndrome_coronavirus_stra...   \n",
       "21  Rabies_lyssavirus_isolate_19RS359_complete_genome   \n",
       "22                             RaTG13_complete_genome   \n",
       "23          Cloning_vector_pcDNA3_1_complete_sequence   \n",
       "24  Friend_murine_leukemia_virus_FB29_complete_genome   \n",
       "25                 Japanese_encephalitis_virus_genome   \n",
       "26  Avian_infectious_bronchitis_virus_complete_genome   \n",
       "27                 Human_adenovirus_A_complete_genome   \n",
       "28                     Dengue_virus_2_complete_genome   \n",
       "29                     Dengue_virus_1_complete_genome   \n",
       "30       Moloney_murine_sarcoma_virus_complete_genome   \n",
       "31      Simian_immunodeficiency_virus_complete_genome   \n",
       "32  Marburg_marburgvirus_isolate_Marburg_virus-H.s...   \n",
       "33  African_swine_fever_virus_strain_BA71V_complet...   \n",
       "34                    Simian_virus_40_complete_genome   \n",
       "35                       Hendra_virus_complete_genome   \n",
       "36             canine_distemper_virus_complete_genome   \n",
       "37  Feline_infectious_peritonitis_virus_complete_g...   \n",
       "38             Human_coronavirus_229E_complete_genome   \n",
       "39                        Nipah_virus_complete_genome   \n",
       "40    Porcine_epidemic_diarrhea_virus_complete_genome   \n",
       "41                       Cowpox_virus_complete_genome   \n",
       "42               Goatpox_virus_Pellor_complete_genome   \n",
       "43       Hepatitis_C_virus_genotype_1_complete_genome   \n",
       "44                  Chikungunya_virus_complete_genome   \n",
       "45              SARS_coronavirus_Tor2_complete_genome   \n",
       "46  Crimean-Congo_hemorrhagic_fever_virus_segment_...   \n",
       "47  Crimean-Congo_hemorrhagic_fever_virus_segment_...   \n",
       "48  Crimean-Congo_hemorrhagic_fever_virus_segment_...   \n",
       "49  Human_coronavirus_OC43_strain_ATCC_VR-759_comp...   \n",
       "50  Human_herpesvirus_5_strain_Merlin_complete_genome   \n",
       "51       Singapore_grouper_iridovirus_complete_genome   \n",
       "52                               HKU4_complete_genome   \n",
       "53                               HKU5_complete_genome   \n",
       "54                         Zika_virus_complete_genome   \n",
       "55  Middle_East_respiratory_syndrome-related_coron...   \n",
       "56  Hubei_picorna-like_virus_45_strain_spider13352...   \n",
       "57  Vesicular_stomatitis_Indiana_virus_strain_98CO...   \n",
       "58  UNVERIFIED_African_swine_fever_virus_isolate_O...   \n",
       "59              SARS_CoV_2_Wuahn-Hu-1_complete_genome   \n",
       "60  Murine_hepatitis_virus_strain_A59_complete_genome   \n",
       "61  Avian_avulavirus_1_isolate_chicken-Indonesia-I...   \n",
       "62     Bat_coronavirus_BtCoV-133-2005_complete_genome   \n",
       "63  Tylonycteris_pachypus_bat_coronavirus_HKU4-rel...   \n",
       "64                BtPa-BetaCoV_GD2013_complete_genome   \n",
       "65  Pipistrellus_abramus_bat_coronavirus_HKU5-rela...   \n",
       "66  Homo_sapiens_lncAB371_6_lncRNA_gene_complete_s...   \n",
       "67  Homo_sapiens_lncAB370_3_lncRNA_gene_complete_s...   \n",
       "68       Homo_sapiens_clone_BAC_JH12_genomic_sequence   \n",
       "69  Homo_sapiens_lncAB599_3_lncRNA_gene_complete_s...   \n",
       "70  Rattus_norvegicus_4_BAC_CH230-209E1_Childrens_...   \n",
       "71  Pig_DNA_sequence_from_clone_CH242-68L20_on_chr...   \n",
       "72  Mus_musculus_targeted_non-conditional_lacZ-tag...   \n",
       "73  Mus_musculus_BAC_clone_RP23-168H19_from_10_com...   \n",
       "74                     J01749_1_cloning_vector_pBR322   \n",
       "75                KY926444_1_cloning_vector_pET-hsp12   \n",
       "76                 MK787297_1_cloning_vector_pR6K-KCA   \n",
       "77       MK908408_1_cloning_vector_pUC_3GLA_U6_1_Bbsl   \n",
       "78        MN996867_1_cloning_vector_pcDNA3_1_complete   \n",
       "79                   U13856_1_pGEX-5X1_cloning_vector   \n",
       "80                     X65300_1_cloning_vector_pGEM_1   \n",
       "81                          NC_000001_11_RPS6KA1_gene   \n",
       "82                            NC_000004_12_ABCG2_gene   \n",
       "83                             NC_000004_12_FGF2_gene   \n",
       "84                             NC_000005_10_TERT_gene   \n",
       "85                            NC_000006_12_VEGFA_gene   \n",
       "86                             NC_000007_14_CAV1_gene   \n",
       "87                             NC_000007_14_CFTR_gene   \n",
       "88                            NC_000007_14_KCNH2_gene   \n",
       "89                            NC_000017_11_BIRC5_gene   \n",
       "90                             NC_000017_11_TP53_gene   \n",
       "91                              NC_000019_10_BAX_gene   \n",
       "92                              NC_000021_9_SOD1_gene   \n",
       "93                             NC_000068_8_Cebpb_gene   \n",
       "94                              NC_000069_7_Adar_gene   \n",
       "95                               NC_000074_7_Mt1_gene   \n",
       "96                             NC_000076_7_Ascl1_gene   \n",
       "\n",
       "                    Accession  \\\n",
       "0                  AF274751_1   \n",
       "1                  AJ289709_1   \n",
       "2                  DQ011855_1   \n",
       "3                  DQ022305_2   \n",
       "4                  DQ079883_1   \n",
       "5                  DQ079895_1   \n",
       "6                  DQ084199_1   \n",
       "7                  DQ084200_1   \n",
       "8                  EU410304_1   \n",
       "9                  FJ217162_1   \n",
       "10                 JF714652_1   \n",
       "11                 KC881006_1   \n",
       "12                 KF367457_1   \n",
       "13                 KF686346_1   \n",
       "14                 KJ997912_1   \n",
       "15                 KX868289_2   \n",
       "16                   L22089_1   \n",
       "17                   M19183_1   \n",
       "18                 MG772933_1   \n",
       "19                 MK625641_1   \n",
       "20                 MK651076_1   \n",
       "21                 MK920923_1   \n",
       "22                 MN996532_1   \n",
       "23                 MN996867_1   \n",
       "24                NC_001362_1   \n",
       "25                NC_001437_1   \n",
       "26                NC_001451_1   \n",
       "27                NC_001460_1   \n",
       "28                NC_001474_2   \n",
       "29                NC_001477_1   \n",
       "30                NC_001502_1   \n",
       "31                NC_001549_1   \n",
       "32                NC_001608_3   \n",
       "33                NC_001659_2   \n",
       "34                NC_001669_1   \n",
       "35                NC_001906_3   \n",
       "36                NC_001921_1   \n",
       "37                NC_002306_3   \n",
       "38                NC_002645_1   \n",
       "39                NC_002728_1   \n",
       "40                NC_003436_1   \n",
       "41                NC_003663_2   \n",
       "42                NC_004003_1   \n",
       "43                NC_004102_1   \n",
       "44                NC_004162_2   \n",
       "45                NC_004718_3   \n",
       "46                NC_005300_2   \n",
       "47                NC_005301_3   \n",
       "48                NC_005302_1   \n",
       "49                NC_006213_1   \n",
       "50                NC_006273_2   \n",
       "51                NC_006549_1   \n",
       "52                NC_009019_1   \n",
       "53                NC_009020_1   \n",
       "54                NC_012532_1   \n",
       "55                NC_019843_3   \n",
       "56                NC_032993_1   \n",
       "57                NC_038236_1   \n",
       "58                NC_044948_1   \n",
       "59                NC_045512_2   \n",
       "60                NC_048217_1   \n",
       "61                 MN727300_2   \n",
       "62                 DQ648794_1   \n",
       "63                 MN611519_1   \n",
       "64                 KJ473820_1   \n",
       "65                 MN611520_1   \n",
       "66                 MK280367_1   \n",
       "67                 MK280359_1   \n",
       "68                 MF164268_1   \n",
       "69                 MK279923_1   \n",
       "70                 AC105600_5   \n",
       "71                 FP326762_5   \n",
       "72                 JN950623_1   \n",
       "73                 AC123074_2   \n",
       "74                   J01749_1   \n",
       "75                 KY926444_1   \n",
       "76                 MK787297_1   \n",
       "77                 MK908408_1   \n",
       "78                 MN996867_1   \n",
       "79                   U13856_1   \n",
       "80                   X65300_1   \n",
       "81  NC_000001_11_RPS6KA1_gene   \n",
       "82    NC_000004_12_ABCG2_gene   \n",
       "83     NC_000004_12_FGF2_gene   \n",
       "84     NC_000005_10_TERT_gene   \n",
       "85    NC_000006_12_VEGFA_gene   \n",
       "86     NC_000007_14_CAV1_gene   \n",
       "87     NC_000007_14_CFTR_gene   \n",
       "88    NC_000007_14_KCNH2_gene   \n",
       "89    NC_000017_11_BIRC5_gene   \n",
       "90     NC_000017_11_TP53_gene   \n",
       "91      NC_000019_10_BAX_gene   \n",
       "92      NC_000021_9_SOD1_gene   \n",
       "93     NC_000068_8_Cebpb_gene   \n",
       "94      NC_000069_7_Adar_gene   \n",
       "95       NC_000074_7_Mt1_gene   \n",
       "96     NC_000076_7_Ascl1_gene   \n",
       "\n",
       "                                            File_Name  \\\n",
       "0     AF274751_1_Bacteriophage_S13_complete_genome.fa   \n",
       "1   AJ289709_1_Human_endogenous_retrovirus_H_HERV-...   \n",
       "2   DQ011855_1_Porcine_hemagglutinating_encephalom...   \n",
       "3   DQ022305_2_Bat_SARS_coronavirus HKU3-1_complet...   \n",
       "4        DQ079883_1_Coliphage_ID45_complete_genome.fa   \n",
       "5        DQ079895_1_Coliphage_WA11_complete_genome.fa   \n",
       "6   DQ084199_1_bat_SARS_coronavirus_HKU3-2_complet...   \n",
       "7   DQ084200_1_bat_SARS_coronavirus_HKU3-3_complet...   \n",
       "8   EU410304_1_Synthetic_Vaccinia_virus_clone_GLV-...   \n",
       "9   FJ217162_1_Cote_d-Ivoire_ebolavirus_complete_g...   \n",
       "10  JF714652_1_XMRV-like_mouse_endogenous_retrovir...   \n",
       "11  KC881006_1_Bat_SARS-like_coronavirus_Rs3367_co...   \n",
       "12  KF367457_1_Bat_SARS-like_coronavirus_WIV1_comp...   \n",
       "13  KF686346_1_Human_coronavirus_HKU1_strain_HKU1-...   \n",
       "14  KJ997912_1_Enterobacteria_phage_MED1_complete_...   \n",
       "15  KX868289_2_Human_adenovirus_12_isolate_GyK010_...   \n",
       "16  L22089_1_Mengo_virus_isolate_M_complete_genome.fa   \n",
       "17  M19183_1_Woodchuck_hepatitis_virus_WHV_complet...   \n",
       "18  MG772933_1_Bat_SARS-like_coronavirus_isolate_b...   \n",
       "19  MK625641_1_Porcine_deltacoronavirus_isolate_CH...   \n",
       "20  MK651076_1_Swine_acute_diarrhea_syndrome_coron...   \n",
       "21  MK920923_1_Rabies_lyssavirus_isolate_19RS359_c...   \n",
       "22               MN996532_1_RaTG13_complete_genome.fa   \n",
       "23  MN996867_1_Cloning_vector_pcDNA3_1_complete_se...   \n",
       "24  NC_001362_1_Friend_murine_leukemia_virus_FB29_...   \n",
       "25  NC_001437_1_Japanese_encephalitis_virus_genome.fa   \n",
       "26  NC_001451_1_Avian_infectious_bronchitis_virus_...   \n",
       "27  NC_001460_1_Human_adenovirus_A_complete_genome.fa   \n",
       "28      NC_001474_2_Dengue_virus_2_complete_genome.fa   \n",
       "29      NC_001477_1_Dengue_virus_1_complete_genome.fa   \n",
       "30  NC_001502_1_Moloney_murine_sarcoma_virus_compl...   \n",
       "31  NC_001549_1_Simian_immunodeficiency_virus_comp...   \n",
       "32  NC_001608_3_Marburg_marburgvirus_isolate_Marbu...   \n",
       "33  NC_001659_2_African_swine_fever_virus_strain_B...   \n",
       "34     NC_001669_1_Simian_virus_40_complete_genome.fa   \n",
       "35        NC_001906_3_Hendra_virus_complete_genome.fa   \n",
       "36  NC_001921_1_canine_distemper_virus_complete_ge...   \n",
       "37  NC_002306_3_Feline_infectious_peritonitis_viru...   \n",
       "38  NC_002645_1_Human_coronavirus_229E_complete_ge...   \n",
       "39         NC_002728_1_Nipah_virus_complete_genome.fa   \n",
       "40  NC_003436_1_Porcine_epidemic_diarrhea_virus_co...   \n",
       "41        NC_003663_2_Cowpox_virus_complete_genome.fa   \n",
       "42  NC_004003_1_Goatpox_virus_Pellor_complete_geno...   \n",
       "43  NC_004102_1_Hepatitis_C_virus_genotype_1_compl...   \n",
       "44   NC_004162_2_Chikungunya_virus_complete_genome.fa   \n",
       "45  NC_004718_3_SARS_coronavirus_Tor2_complete_gen...   \n",
       "46  NC_005300_2_Crimean-Congo_hemorrhagic_fever_vi...   \n",
       "47  NC_005301_3_Crimean-Congo_hemorrhagic_fever_vi...   \n",
       "48  NC_005302_1_Crimean-Congo_hemorrhagic_fever_vi...   \n",
       "49  NC_006213_1_Human_coronavirus_OC43_strain_ATCC...   \n",
       "50  NC_006273_2_Human_herpesvirus_5_strain_Merlin_...   \n",
       "51  NC_006549_1_Singapore_grouper_iridovirus_compl...   \n",
       "52                NC_009019_1_HKU4_complete_genome.fa   \n",
       "53                NC_009020_1_HKU5_complete_genome.fa   \n",
       "54          NC_012532_1_Zika_virus_complete_genome.fa   \n",
       "55  NC_019843_3_Middle_East_respiratory_syndrome-r...   \n",
       "56  NC_032993_1_Hubei_picorna-like_virus_45_strain...   \n",
       "57  NC_038236_1_Vesicular_stomatitis_Indiana_virus...   \n",
       "58  NC_044948_1_UNVERIFIED_African_swine_fever_vir...   \n",
       "59  NC_045512_2_SARS_CoV_2_Wuahn-Hu-1_complete_gen...   \n",
       "60  NC_048217_1_Murine_hepatitis_virus_strain_A59_...   \n",
       "61  MN727300_2_Avian_avulavirus_1_isolate_chicken-...   \n",
       "62  DQ648794_1_Bat_coronavirus_BtCoV-133-2005_comp...   \n",
       "63  MN611519_1_Tylonycteris_pachypus_bat_coronavir...   \n",
       "64  KJ473820_1_BtPa-BetaCoV_GD2013_complete_genome.fa   \n",
       "65  MN611520_1_Pipistrellus_abramus_bat_coronaviru...   \n",
       "66  MK280367_1_Homo_sapiens_lncAB371_6_lncRNA_gene...   \n",
       "67  MK280359_1_Homo_sapiens_lncAB370_3_lncRNA_gene...   \n",
       "68  MF164268_1_Homo_sapiens_clone_BAC_JH12_genomic...   \n",
       "69  MK279923_1_Homo_sapiens_lncAB599_3_lncRNA_gene...   \n",
       "70  AC105600_5_Rattus_norvegicus_4_BAC_CH230-209E1...   \n",
       "71  FP326762_5_Pig_DNA_sequence_from_clone_CH242-6...   \n",
       "72  JN950623_1_Mus_musculus_targeted_non-condition...   \n",
       "73  AC123074_2_Mus_musculus_BAC_clone_RP23-168H19_...   \n",
       "74                  J01749_1_cloning_vector_pBR322.fa   \n",
       "75             KY926444_1_cloning_vector_pET-hsp12.fa   \n",
       "76              MK787297_1_cloning_vector_pR6K-KCA.fa   \n",
       "77    MK908408_1_cloning_vector_pUC_3GLA_U6_1_Bbsl.fa   \n",
       "78     MN996867_1_cloning_vector_pcDNA3_1_complete.fa   \n",
       "79                U13856_1_pGEX-5X1_cloning_vector.fa   \n",
       "80                  X65300_1_cloning_vector_pGEM_1.fa   \n",
       "81                       NC_000001_11_RPS6KA1_gene.fa   \n",
       "82                         NC_000004_12_ABCG2_gene.fa   \n",
       "83                          NC_000004_12_FGF2_gene.fa   \n",
       "84                          NC_000005_10_TERT_gene.fa   \n",
       "85                         NC_000006_12_VEGFA_gene.fa   \n",
       "86                          NC_000007_14_CAV1_gene.fa   \n",
       "87                          NC_000007_14_CFTR_gene.fa   \n",
       "88                         NC_000007_14_KCNH2_gene.fa   \n",
       "89                         NC_000017_11_BIRC5_gene.fa   \n",
       "90                          NC_000017_11_TP53_gene.fa   \n",
       "91                           NC_000019_10_BAX_gene.fa   \n",
       "92                           NC_000021_9_SOD1_gene.fa   \n",
       "93                          NC_000068_8_Cebpb_gene.fa   \n",
       "94                           NC_000069_7_Adar_gene.fa   \n",
       "95                            NC_000074_7_Mt1_gene.fa   \n",
       "96                          NC_000076_7_Ascl1_gene.fa   \n",
       "\n",
       "                      category  \n",
       "0                 BSL_organism  \n",
       "1                 BSL_organism  \n",
       "2                 BSL_organism  \n",
       "3                 BSL_organism  \n",
       "4                 BSL_organism  \n",
       "5                 BSL_organism  \n",
       "6                 BSL_organism  \n",
       "7                 BSL_organism  \n",
       "8                 BSL_organism  \n",
       "9                 BSL_organism  \n",
       "10                BSL_organism  \n",
       "11                BSL_organism  \n",
       "12                BSL_organism  \n",
       "13                BSL_organism  \n",
       "14                BSL_organism  \n",
       "15                BSL_organism  \n",
       "16                BSL_organism  \n",
       "17                BSL_organism  \n",
       "18                BSL_organism  \n",
       "19                BSL_organism  \n",
       "20                BSL_organism  \n",
       "21                BSL_organism  \n",
       "22                BSL_organism  \n",
       "23                BSL_organism  \n",
       "24                BSL_organism  \n",
       "25                BSL_organism  \n",
       "26                BSL_organism  \n",
       "27                BSL_organism  \n",
       "28                BSL_organism  \n",
       "29                BSL_organism  \n",
       "30                BSL_organism  \n",
       "31                BSL_organism  \n",
       "32                BSL_organism  \n",
       "33                BSL_organism  \n",
       "34                BSL_organism  \n",
       "35                BSL_organism  \n",
       "36                BSL_organism  \n",
       "37                BSL_organism  \n",
       "38                BSL_organism  \n",
       "39                BSL_organism  \n",
       "40                BSL_organism  \n",
       "41                BSL_organism  \n",
       "42                BSL_organism  \n",
       "43                BSL_organism  \n",
       "44                BSL_organism  \n",
       "45                BSL_organism  \n",
       "46                BSL_organism  \n",
       "47                BSL_organism  \n",
       "48                BSL_organism  \n",
       "49                BSL_organism  \n",
       "50                BSL_organism  \n",
       "51                BSL_organism  \n",
       "52                BSL_organism  \n",
       "53                BSL_organism  \n",
       "54                BSL_organism  \n",
       "55                BSL_organism  \n",
       "56                BSL_organism  \n",
       "57                BSL_organism  \n",
       "58                BSL_organism  \n",
       "59                BSL_organism  \n",
       "60                BSL_organism  \n",
       "61                BSL_organism  \n",
       "62                     bat_cov  \n",
       "63                     bat_cov  \n",
       "64                     bat_cov  \n",
       "65                     bat_cov  \n",
       "66              mamalian_genes  \n",
       "67              mamalian_genes  \n",
       "68              mamalian_genes  \n",
       "69              mamalian_genes  \n",
       "70              mamalian_genes  \n",
       "71              mamalian_genes  \n",
       "72              mamalian_genes  \n",
       "73              mamalian_genes  \n",
       "74              cloning_vector  \n",
       "75              cloning_vector  \n",
       "76              cloning_vector  \n",
       "77              cloning_vector  \n",
       "78              cloning_vector  \n",
       "79              cloning_vector  \n",
       "80              cloning_vector  \n",
       "81  mamalian_expression_vector  \n",
       "82  mamalian_expression_vector  \n",
       "83  mamalian_expression_vector  \n",
       "84  mamalian_expression_vector  \n",
       "85  mamalian_expression_vector  \n",
       "86  mamalian_expression_vector  \n",
       "87  mamalian_expression_vector  \n",
       "88  mamalian_expression_vector  \n",
       "89  mamalian_expression_vector  \n",
       "90  mamalian_expression_vector  \n",
       "91  mamalian_expression_vector  \n",
       "92  mamalian_expression_vector  \n",
       "93  mamalian_expression_vector  \n",
       "94  mamalian_expression_vector  \n",
       "95  mamalian_expression_vector  \n",
       "96  mamalian_expression_vector  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.options.display.max_rows = 2000\n",
    "df=pd.read_csv(ORGANISM_CSVS[0])\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "danish-algebra",
   "metadata": {},
   "source": [
    "## Bat Spikes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "marine-performance",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running blast2 on SRR5885860\n",
      "Elapsed: 43.773359298706055\n"
     ]
    }
   ],
   "source": [
    "bat_result_list=main_worflow(ORGANISM_CSVS[1],CSV_PATHS[1], search_name='batcov', rebuild_db=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "dirty-collar",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'SRR5885860': []}]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bat_result_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "arabic-height",
   "metadata": {},
   "source": [
    "### Bat spikes and misc other virueses searching for in second run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "about-honolulu",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Accession</th>\n",
       "      <th>File_Name</th>\n",
       "      <th>category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAP13441_1_SARS-CoV_Urbani_apike</td>\n",
       "      <td>AAP13441_1_SARS-CoV_Urbani_apike</td>\n",
       "      <td>AAP13441_1_SARS-CoV_Urbani_apike.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AB981587_1_Influenza_virus_type_A_A-NWS-33_H1N...</td>\n",
       "      <td>AB981587_1_Influenza_virus_type_A_A-NWS-33_H1N...</td>\n",
       "      <td>AB981587_1_Influenza_virus_type_A_A-NWS-33_H1N...</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AcCoV-JC34_spike_YP_009380521</td>\n",
       "      <td>AcCoV-JC34_spike_YP_009380521</td>\n",
       "      <td>AcCoV-JC34_spike_YP_009380521.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BM48-31_BGR_2008</td>\n",
       "      <td>BM48-31_BGR_2008</td>\n",
       "      <td>BM48-31_BGR_2008.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BtRl-BetaCoV_SC2018</td>\n",
       "      <td>BtRl-BetaCoV_SC2018</td>\n",
       "      <td>BtRl-BetaCoV_SC2018.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BtRs-BetaCoV_HuB2013</td>\n",
       "      <td>BtRs-BetaCoV_HuB2013</td>\n",
       "      <td>BtRs-BetaCoV_HuB2013.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>civet014_AAU04661</td>\n",
       "      <td>civet014_AAU04661</td>\n",
       "      <td>civet014_AAU04661.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CoV_ZC45</td>\n",
       "      <td>CoV_ZC45</td>\n",
       "      <td>CoV_ZC45.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CoV_ZXC21</td>\n",
       "      <td>CoV_ZXC21</td>\n",
       "      <td>CoV_ZXC21.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CoV_ZXC21_</td>\n",
       "      <td>CoV_ZXC21_</td>\n",
       "      <td>CoV_ZXC21_.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GD03_AAS10463</td>\n",
       "      <td>GD03_AAS10463</td>\n",
       "      <td>GD03_AAS10463.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HKU3-1_AAY88866</td>\n",
       "      <td>HKU3-1_AAY88866</td>\n",
       "      <td>HKU3-1_AAY88866.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HKU3_Spike_mouse_adapted_RBD_QND76034</td>\n",
       "      <td>HKU3_Spike_mouse_adapted_RBD_QND76034</td>\n",
       "      <td>HKU3_Spike_mouse_adapted_RBD_QND76034.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HQ008256_1_Influenza_virus_type_A_A-NWS-33_H1N...</td>\n",
       "      <td>HQ008256_1_Influenza_virus_type_A_A-NWS-33_H1N...</td>\n",
       "      <td>HQ008256_1_Influenza_virus_type_A_A-NWS-33_H1N...</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HQ008527_1_Influenza_virus_type_A_A-NWS-33_H1N...</td>\n",
       "      <td>HQ008527_1_Influenza_virus_type_A_A-NWS-33_H1N...</td>\n",
       "      <td>HQ008527_1_Influenza_virus_type_A_A-NWS-33_H1N...</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Longquan-140_spike_P_AID16716-1</td>\n",
       "      <td>Longquan-140_spike_P_AID16716-1</td>\n",
       "      <td>Longquan-140_spike_P_AID16716-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>M91479_1_Influenza_virus_type_A_A-NWS-33_H1N1_...</td>\n",
       "      <td>M91479_1_Influenza_virus_type_A_A-NWS-33_H1N1_...</td>\n",
       "      <td>M91479_1_Influenza_virus_type_A_A-NWS-33_H1N1_...</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MERS_Spike_QBM11748</td>\n",
       "      <td>MERS_Spike_QBM11748</td>\n",
       "      <td>MERS_Spike_QBM11748.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NC_002645_1_Human_coronavirus_229E_complete_ge...</td>\n",
       "      <td>NC_002645_1_Human_coronavirus_229E_complete_ge...</td>\n",
       "      <td>NC_002645_1_Human_coronavirus_229E_complete_ge...</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>OC43_spike_glycoprotein_P36334_1</td>\n",
       "      <td>OC43_spike_glycoprotein_P36334_1</td>\n",
       "      <td>OC43_spike_glycoprotein_P36334_1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PC4-115_AAV49719_spike</td>\n",
       "      <td>PC4-115_AAV49719_spike</td>\n",
       "      <td>PC4-115_AAV49719_spike.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>RaT13_spike-closed_pdb_6ZGF_C</td>\n",
       "      <td>RaT13_spike-closed_pdb_6ZGF_C</td>\n",
       "      <td>RaT13_spike-closed_pdb_6ZGF_C.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>RaT13_spike_QHR63300_2</td>\n",
       "      <td>RaT13_spike_QHR63300_2</td>\n",
       "      <td>RaT13_spike_QHR63300_2.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>RaTG13</td>\n",
       "      <td>RaTG13</td>\n",
       "      <td>RaTG13.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>RaTG13_MN996532_2_complete</td>\n",
       "      <td>RaTG13_MN996532_2_complete</td>\n",
       "      <td>RaTG13_MN996532_2_complete.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Rf4092_spike_P_ATO98145-1</td>\n",
       "      <td>Rf4092_spike_P_ATO98145-1</td>\n",
       "      <td>Rf4092_spike_P_ATO98145-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>RmYN02_QPD89843_partial_splike</td>\n",
       "      <td>RmYN02_QPD89843_partial_splike</td>\n",
       "      <td>RmYN02_QPD89843_partial_splike.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>RP3_AAZ67052</td>\n",
       "      <td>RP3_AAZ67052</td>\n",
       "      <td>RP3_AAZ67052.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>RP3_AAZ67052_spike</td>\n",
       "      <td>RP3_AAZ67052_spike</td>\n",
       "      <td>RP3_AAZ67052_spike.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Rs3367_spike_P_AGZ48818-1</td>\n",
       "      <td>Rs3367_spike_P_AGZ48818-1</td>\n",
       "      <td>Rs3367_spike_P_AGZ48818-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Rs4081_spike_P_ATO98120-1</td>\n",
       "      <td>Rs4081_spike_P_ATO98120-1</td>\n",
       "      <td>Rs4081_spike_P_ATO98120-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Rs4084_spike_P_ATO98132-1</td>\n",
       "      <td>Rs4084_spike_P_ATO98132-1</td>\n",
       "      <td>Rs4084_spike_P_ATO98132-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Rs4231_spike_P_ATO98157-1</td>\n",
       "      <td>Rs4231_spike_P_ATO98157-1</td>\n",
       "      <td>Rs4231_spike_P_ATO98157-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Rs4237_spike_P_ATO98169-1</td>\n",
       "      <td>Rs4237_spike_P_ATO98169-1</td>\n",
       "      <td>Rs4237_spike_P_ATO98169-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Rs4247_spike_P_ATO98181-1</td>\n",
       "      <td>Rs4247_spike_P_ATO98181-1</td>\n",
       "      <td>Rs4247_spike_P_ATO98181-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Rs4255_spike_P_ATO98193-1</td>\n",
       "      <td>Rs4255_spike_P_ATO98193-1</td>\n",
       "      <td>Rs4255_spike_P_ATO98193-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Rs4874_spike_P_ATO98205-1</td>\n",
       "      <td>Rs4874_spike_P_ATO98205-1</td>\n",
       "      <td>Rs4874_spike_P_ATO98205-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Rs5567_spike_P_AGZ48818-1</td>\n",
       "      <td>Rs5567_spike_P_AGZ48818-1</td>\n",
       "      <td>Rs5567_spike_P_AGZ48818-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Rs7327_spike_P_ATO98218-1</td>\n",
       "      <td>Rs7327_spike_P_ATO98218-1</td>\n",
       "      <td>Rs7327_spike_P_ATO98218-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Rs9401_spike_P_ATO98231-1</td>\n",
       "      <td>Rs9401_spike_P_ATO98231-1</td>\n",
       "      <td>Rs9401_spike_P_ATO98231-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>RsSHC014</td>\n",
       "      <td>RsSHC014</td>\n",
       "      <td>RsSHC014.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>RsSHC014_spike_P_AGZ48806-1</td>\n",
       "      <td>RsSHC014_spike_P_AGZ48806-1</td>\n",
       "      <td>RsSHC014_spike_P_AGZ48806-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>SARS-CoV-2_reference_genome</td>\n",
       "      <td>SARS-CoV-2_reference_genome</td>\n",
       "      <td>SARS-CoV-2_reference_genome.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>SARS-CoV-2_spike_glycoprotein_reference_YP_009...</td>\n",
       "      <td>SARS-CoV-2_spike_glycoprotein_reference_YP_009...</td>\n",
       "      <td>SARS-CoV-2_spike_glycoprotein_reference_YP_009...</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>SARS-CoV-2_spike_glycoptorein_QHD43416_1</td>\n",
       "      <td>SARS-CoV-2_spike_glycoptorein_QHD43416_1</td>\n",
       "      <td>SARS-CoV-2_spike_glycoptorein_QHD43416_1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>SARS-CoV-2_Spike_QOH25833</td>\n",
       "      <td>SARS-CoV-2_Spike_QOH25833</td>\n",
       "      <td>SARS-CoV-2_Spike_QOH25833.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>SARS-CoV-2_spike_Uniprot_P0DTC2</td>\n",
       "      <td>SARS-CoV-2_spike_Uniprot_P0DTC2</td>\n",
       "      <td>SARS-CoV-2_spike_Uniprot_P0DTC2.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>SARS_GZ02_Spike_AAS00003</td>\n",
       "      <td>SARS_GZ02_Spike_AAS00003</td>\n",
       "      <td>SARS_GZ02_Spike_AAS00003.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>SARS_HTVR_Spike_Follis_et_al_2006</td>\n",
       "      <td>SARS_HTVR_Spike_Follis_et_al_2006</td>\n",
       "      <td>SARS_HTVR_Spike_Follis_et_al_2006.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>SARS_SLLR_Spike_Follis_et_al_2006</td>\n",
       "      <td>SARS_SLLR_Spike_Follis_et_al_2006</td>\n",
       "      <td>SARS_SLLR_Spike_Follis_et_al_2006.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>SZ3_P59594_spike</td>\n",
       "      <td>SZ3_P59594_spike</td>\n",
       "      <td>SZ3_P59594_spike.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>U08903_1_Influenza_virus_type_A_A-NWS-33_H1N1_...</td>\n",
       "      <td>U08903_1_Influenza_virus_type_A_A-NWS-33_H1N1_...</td>\n",
       "      <td>U08903_1_Influenza_virus_type_A_A-NWS-33_H1N1_...</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>U08904_1_Influenza_virus_type_A_A-NWS-33_H1N1_...</td>\n",
       "      <td>U08904_1_Influenza_virus_type_A_A-NWS-33_H1N1_...</td>\n",
       "      <td>U08904_1_Influenza_virus_type_A_A-NWS-33_H1N1_...</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Urbani_AYV99817_spike</td>\n",
       "      <td>Urbani_AYV99817_spike</td>\n",
       "      <td>Urbani_AYV99817_spike.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>WIV16_spike_P_ALK02457-1</td>\n",
       "      <td>WIV16_spike_P_ALK02457-1</td>\n",
       "      <td>WIV16_spike_P_ALK02457-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>WIV1_spike_P_AGZ48828-1</td>\n",
       "      <td>WIV1_spike_P_AGZ48828-1</td>\n",
       "      <td>WIV1_spike_P_AGZ48828-1.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>ZC45_Spike_AVP78031</td>\n",
       "      <td>ZC45_Spike_AVP78031</td>\n",
       "      <td>ZC45_Spike_AVP78031.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>ZXC21_Spike_AVP78042</td>\n",
       "      <td>ZXC21_Spike_AVP78042</td>\n",
       "      <td>ZXC21_Spike_AVP78042.fa</td>\n",
       "      <td>misc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Name  \\\n",
       "0                    AAP13441_1_SARS-CoV_Urbani_apike   \n",
       "1   AB981587_1_Influenza_virus_type_A_A-NWS-33_H1N...   \n",
       "2                       AcCoV-JC34_spike_YP_009380521   \n",
       "3                                    BM48-31_BGR_2008   \n",
       "4                                 BtRl-BetaCoV_SC2018   \n",
       "5                                BtRs-BetaCoV_HuB2013   \n",
       "6                                   civet014_AAU04661   \n",
       "7                                            CoV_ZC45   \n",
       "8                                           CoV_ZXC21   \n",
       "9                                          CoV_ZXC21_   \n",
       "10                                      GD03_AAS10463   \n",
       "11                                    HKU3-1_AAY88866   \n",
       "12              HKU3_Spike_mouse_adapted_RBD_QND76034   \n",
       "13  HQ008256_1_Influenza_virus_type_A_A-NWS-33_H1N...   \n",
       "14  HQ008527_1_Influenza_virus_type_A_A-NWS-33_H1N...   \n",
       "15                    Longquan-140_spike_P_AID16716-1   \n",
       "16  M91479_1_Influenza_virus_type_A_A-NWS-33_H1N1_...   \n",
       "17                                MERS_Spike_QBM11748   \n",
       "18  NC_002645_1_Human_coronavirus_229E_complete_ge...   \n",
       "19                   OC43_spike_glycoprotein_P36334_1   \n",
       "20                             PC4-115_AAV49719_spike   \n",
       "21                      RaT13_spike-closed_pdb_6ZGF_C   \n",
       "22                             RaT13_spike_QHR63300_2   \n",
       "23                                             RaTG13   \n",
       "24                         RaTG13_MN996532_2_complete   \n",
       "25                          Rf4092_spike_P_ATO98145-1   \n",
       "26                     RmYN02_QPD89843_partial_splike   \n",
       "27                                       RP3_AAZ67052   \n",
       "28                                 RP3_AAZ67052_spike   \n",
       "29                          Rs3367_spike_P_AGZ48818-1   \n",
       "30                          Rs4081_spike_P_ATO98120-1   \n",
       "31                          Rs4084_spike_P_ATO98132-1   \n",
       "32                          Rs4231_spike_P_ATO98157-1   \n",
       "33                          Rs4237_spike_P_ATO98169-1   \n",
       "34                          Rs4247_spike_P_ATO98181-1   \n",
       "35                          Rs4255_spike_P_ATO98193-1   \n",
       "36                          Rs4874_spike_P_ATO98205-1   \n",
       "37                          Rs5567_spike_P_AGZ48818-1   \n",
       "38                          Rs7327_spike_P_ATO98218-1   \n",
       "39                          Rs9401_spike_P_ATO98231-1   \n",
       "40                                           RsSHC014   \n",
       "41                        RsSHC014_spike_P_AGZ48806-1   \n",
       "42                        SARS-CoV-2_reference_genome   \n",
       "43  SARS-CoV-2_spike_glycoprotein_reference_YP_009...   \n",
       "44           SARS-CoV-2_spike_glycoptorein_QHD43416_1   \n",
       "45                          SARS-CoV-2_Spike_QOH25833   \n",
       "46                    SARS-CoV-2_spike_Uniprot_P0DTC2   \n",
       "47                           SARS_GZ02_Spike_AAS00003   \n",
       "48                  SARS_HTVR_Spike_Follis_et_al_2006   \n",
       "49                  SARS_SLLR_Spike_Follis_et_al_2006   \n",
       "50                                   SZ3_P59594_spike   \n",
       "51  U08903_1_Influenza_virus_type_A_A-NWS-33_H1N1_...   \n",
       "52  U08904_1_Influenza_virus_type_A_A-NWS-33_H1N1_...   \n",
       "53                              Urbani_AYV99817_spike   \n",
       "54                           WIV16_spike_P_ALK02457-1   \n",
       "55                            WIV1_spike_P_AGZ48828-1   \n",
       "56                                ZC45_Spike_AVP78031   \n",
       "57                               ZXC21_Spike_AVP78042   \n",
       "\n",
       "                                            Accession  \\\n",
       "0                    AAP13441_1_SARS-CoV_Urbani_apike   \n",
       "1   AB981587_1_Influenza_virus_type_A_A-NWS-33_H1N...   \n",
       "2                       AcCoV-JC34_spike_YP_009380521   \n",
       "3                                    BM48-31_BGR_2008   \n",
       "4                                 BtRl-BetaCoV_SC2018   \n",
       "5                                BtRs-BetaCoV_HuB2013   \n",
       "6                                   civet014_AAU04661   \n",
       "7                                            CoV_ZC45   \n",
       "8                                           CoV_ZXC21   \n",
       "9                                          CoV_ZXC21_   \n",
       "10                                      GD03_AAS10463   \n",
       "11                                    HKU3-1_AAY88866   \n",
       "12              HKU3_Spike_mouse_adapted_RBD_QND76034   \n",
       "13  HQ008256_1_Influenza_virus_type_A_A-NWS-33_H1N...   \n",
       "14  HQ008527_1_Influenza_virus_type_A_A-NWS-33_H1N...   \n",
       "15                    Longquan-140_spike_P_AID16716-1   \n",
       "16  M91479_1_Influenza_virus_type_A_A-NWS-33_H1N1_...   \n",
       "17                                MERS_Spike_QBM11748   \n",
       "18  NC_002645_1_Human_coronavirus_229E_complete_ge...   \n",
       "19                   OC43_spike_glycoprotein_P36334_1   \n",
       "20                             PC4-115_AAV49719_spike   \n",
       "21                      RaT13_spike-closed_pdb_6ZGF_C   \n",
       "22                             RaT13_spike_QHR63300_2   \n",
       "23                                             RaTG13   \n",
       "24                         RaTG13_MN996532_2_complete   \n",
       "25                          Rf4092_spike_P_ATO98145-1   \n",
       "26                     RmYN02_QPD89843_partial_splike   \n",
       "27                                       RP3_AAZ67052   \n",
       "28                                 RP3_AAZ67052_spike   \n",
       "29                          Rs3367_spike_P_AGZ48818-1   \n",
       "30                          Rs4081_spike_P_ATO98120-1   \n",
       "31                          Rs4084_spike_P_ATO98132-1   \n",
       "32                          Rs4231_spike_P_ATO98157-1   \n",
       "33                          Rs4237_spike_P_ATO98169-1   \n",
       "34                          Rs4247_spike_P_ATO98181-1   \n",
       "35                          Rs4255_spike_P_ATO98193-1   \n",
       "36                          Rs4874_spike_P_ATO98205-1   \n",
       "37                          Rs5567_spike_P_AGZ48818-1   \n",
       "38                          Rs7327_spike_P_ATO98218-1   \n",
       "39                          Rs9401_spike_P_ATO98231-1   \n",
       "40                                           RsSHC014   \n",
       "41                        RsSHC014_spike_P_AGZ48806-1   \n",
       "42                        SARS-CoV-2_reference_genome   \n",
       "43  SARS-CoV-2_spike_glycoprotein_reference_YP_009...   \n",
       "44           SARS-CoV-2_spike_glycoptorein_QHD43416_1   \n",
       "45                          SARS-CoV-2_Spike_QOH25833   \n",
       "46                    SARS-CoV-2_spike_Uniprot_P0DTC2   \n",
       "47                           SARS_GZ02_Spike_AAS00003   \n",
       "48                  SARS_HTVR_Spike_Follis_et_al_2006   \n",
       "49                  SARS_SLLR_Spike_Follis_et_al_2006   \n",
       "50                                   SZ3_P59594_spike   \n",
       "51  U08903_1_Influenza_virus_type_A_A-NWS-33_H1N1_...   \n",
       "52  U08904_1_Influenza_virus_type_A_A-NWS-33_H1N1_...   \n",
       "53                              Urbani_AYV99817_spike   \n",
       "54                           WIV16_spike_P_ALK02457-1   \n",
       "55                            WIV1_spike_P_AGZ48828-1   \n",
       "56                                ZC45_Spike_AVP78031   \n",
       "57                               ZXC21_Spike_AVP78042   \n",
       "\n",
       "                                            File_Name category  \n",
       "0                 AAP13441_1_SARS-CoV_Urbani_apike.fa     misc  \n",
       "1   AB981587_1_Influenza_virus_type_A_A-NWS-33_H1N...     misc  \n",
       "2                    AcCoV-JC34_spike_YP_009380521.fa     misc  \n",
       "3                                 BM48-31_BGR_2008.fa     misc  \n",
       "4                              BtRl-BetaCoV_SC2018.fa     misc  \n",
       "5                             BtRs-BetaCoV_HuB2013.fa     misc  \n",
       "6                                civet014_AAU04661.fa     misc  \n",
       "7                                         CoV_ZC45.fa     misc  \n",
       "8                                        CoV_ZXC21.fa     misc  \n",
       "9                                       CoV_ZXC21_.fa     misc  \n",
       "10                                   GD03_AAS10463.fa     misc  \n",
       "11                                 HKU3-1_AAY88866.fa     misc  \n",
       "12           HKU3_Spike_mouse_adapted_RBD_QND76034.fa     misc  \n",
       "13  HQ008256_1_Influenza_virus_type_A_A-NWS-33_H1N...     misc  \n",
       "14  HQ008527_1_Influenza_virus_type_A_A-NWS-33_H1N...     misc  \n",
       "15                 Longquan-140_spike_P_AID16716-1.fa     misc  \n",
       "16  M91479_1_Influenza_virus_type_A_A-NWS-33_H1N1_...     misc  \n",
       "17                             MERS_Spike_QBM11748.fa     misc  \n",
       "18  NC_002645_1_Human_coronavirus_229E_complete_ge...     misc  \n",
       "19                OC43_spike_glycoprotein_P36334_1.fa     misc  \n",
       "20                          PC4-115_AAV49719_spike.fa     misc  \n",
       "21                   RaT13_spike-closed_pdb_6ZGF_C.fa     misc  \n",
       "22                          RaT13_spike_QHR63300_2.fa     misc  \n",
       "23                                          RaTG13.fa     misc  \n",
       "24                      RaTG13_MN996532_2_complete.fa     misc  \n",
       "25                       Rf4092_spike_P_ATO98145-1.fa     misc  \n",
       "26                  RmYN02_QPD89843_partial_splike.fa     misc  \n",
       "27                                    RP3_AAZ67052.fa     misc  \n",
       "28                              RP3_AAZ67052_spike.fa     misc  \n",
       "29                       Rs3367_spike_P_AGZ48818-1.fa     misc  \n",
       "30                       Rs4081_spike_P_ATO98120-1.fa     misc  \n",
       "31                       Rs4084_spike_P_ATO98132-1.fa     misc  \n",
       "32                       Rs4231_spike_P_ATO98157-1.fa     misc  \n",
       "33                       Rs4237_spike_P_ATO98169-1.fa     misc  \n",
       "34                       Rs4247_spike_P_ATO98181-1.fa     misc  \n",
       "35                       Rs4255_spike_P_ATO98193-1.fa     misc  \n",
       "36                       Rs4874_spike_P_ATO98205-1.fa     misc  \n",
       "37                       Rs5567_spike_P_AGZ48818-1.fa     misc  \n",
       "38                       Rs7327_spike_P_ATO98218-1.fa     misc  \n",
       "39                       Rs9401_spike_P_ATO98231-1.fa     misc  \n",
       "40                                        RsSHC014.fa     misc  \n",
       "41                     RsSHC014_spike_P_AGZ48806-1.fa     misc  \n",
       "42                     SARS-CoV-2_reference_genome.fa     misc  \n",
       "43  SARS-CoV-2_spike_glycoprotein_reference_YP_009...     misc  \n",
       "44        SARS-CoV-2_spike_glycoptorein_QHD43416_1.fa     misc  \n",
       "45                       SARS-CoV-2_Spike_QOH25833.fa     misc  \n",
       "46                 SARS-CoV-2_spike_Uniprot_P0DTC2.fa     misc  \n",
       "47                        SARS_GZ02_Spike_AAS00003.fa     misc  \n",
       "48               SARS_HTVR_Spike_Follis_et_al_2006.fa     misc  \n",
       "49               SARS_SLLR_Spike_Follis_et_al_2006.fa     misc  \n",
       "50                                SZ3_P59594_spike.fa     misc  \n",
       "51  U08903_1_Influenza_virus_type_A_A-NWS-33_H1N1_...     misc  \n",
       "52  U08904_1_Influenza_virus_type_A_A-NWS-33_H1N1_...     misc  \n",
       "53                           Urbani_AYV99817_spike.fa     misc  \n",
       "54                        WIV16_spike_P_ALK02457-1.fa     misc  \n",
       "55                         WIV1_spike_P_AGZ48828-1.fa     misc  \n",
       "56                             ZC45_Spike_AVP78031.fa     misc  \n",
       "57                            ZXC21_Spike_AVP78042.fa     misc  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bat=pd.read_csv(ORGANISM_CSVS[1])\n",
    "df_bat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "printable-guess",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
